src/v2i-hub/CARMAStreetsPlugin/src/CARMAStreetsPlugin.cpp

```diff
@@ -77,39 +79,56 @@ void CARMAStreetsPlugin::UpdateConfigSettings() {
 	} 			
 	PLOG(logDEBUG) <<"Kafka producer created";
 
-	if (kafka_conf_consumer->set("bootstrap.servers", kafkaConnectString, error_string)  != RdKafka::Conf::CONF_OK || (kafka_conf_consumer->set("group.id", "streets_group", error_string) != RdKafka::Conf::CONF_OK)) {
+	if (kafka_conf_sp_consumer->set("bootstrap.servers", kafkaConnectString, error_string)  != RdKafka::Conf::CONF_OK
+		 || (kafka_conf_sp_consumer->set("group.id", "streets_group_scheduling_plan", error_string) != RdKafka::Conf::CONF_OK)
+		 || (kafka_conf_spat_consumer->set("bootstrap.servers", kafkaConnectString, error_string)  != RdKafka::Conf::CONF_OK)
+		 || (kafka_conf_spat_consumer->set("group.id", "streets_group_spat", error_string) != RdKafka::Conf::CONF_OK)) {
 		PLOG(logERROR) <<"Setting kafka config group.id options failed with error:" << error_string << "\n" <<"Exiting with exit code 1";
 		exit(1);
 	} else {
 		PLOG(logDEBUG) <<"Kafka config group.id options set successfully";
 	}
-	kafka_conf_consumer->set("enable.partition.eof", "true", error_string);
+	kafka_conf_sp_consumer->set("enable.partition.eof", "true", error_string);
+	kafka_conf_spat_consumer->set("enable.partition.eof", "true", error_string);
 
-	kafka_consumer = RdKafka::KafkaConsumer::create(kafka_conf_consumer, error_string);
-	if ( !kafka_consumer ) {
-		PLOG(logERROR) << "Failed to create Kafka consumer: " << error_string << std::endl;
+	_scheduing_plan_kafka_consumer = RdKafka::KafkaConsumer::create(kafka_conf_sp_consumer, error_string);
+	_spat_kafka_consumer = RdKafka::KafkaConsumer::create(kafka_conf_spat_consumer, error_string);
+
+	if ( !_scheduing_plan_kafka_consumer || !_spat_kafka_consumer) {
+		PLOG(logERROR) << "Failed to create Kafka consumers: " << error_string << std::endl;
 		exit(1);
 	}
-	PLOG(logDEBUG) << "Created consumer " << kafka_consumer->name() << std::endl;
+	PLOG(logDEBUG) << "Created consumer " << _scheduing_plan_kafka_consumer->name() << std::endl;
+	PLOG(logDEBUG) << "Created consumer " << _spat_kafka_consumer->name() << std::endl;
 
-	//create kafka topic
-	RdKafka::Conf *tconf = RdKafka::Conf::create(RdKafka::Conf::CONF_TOPIC);
-	if(!tconf)
+	//create kafka topics
+	RdKafka::Conf *tconf_spat = RdKafka::Conf::create(RdKafka::Conf::CONF_TOPIC);
+	RdKafka::Conf *tconf_sp = RdKafka::Conf::create(RdKafka::Conf::CONF_TOPIC);
+	if(!tconf_spat && !tconf_sp)
 	{
 		PLOG(logERROR) << "RDKafka create topic conf failed ";
 		return;
 	}   
 
-	_topic = RdKafka::Topic::create(kafka_consumer,_subscribeToSchedulingPlanTopic,tconf,error_string);
-	if(!_topic)
+	_scheduing_plan_topic = RdKafka::Topic::create(_scheduing_plan_kafka_consumer,_subscribeToSchedulingPlanTopic,tconf_sp,error_string);
+	if(!_scheduing_plan_topic)
+	{
+		PLOG(logERROR) << "RDKafka create scheduing plan topic failed:" << error_string;
+		return ;
+	}
+
+	_spat_topic = RdKafka::Topic::create(_spat_kafka_consumer,_subscribeToSpatTopic,tconf_spat,error_string);
```

Comment:
do we need to create the topic here or will it already be created by docker-compose setup.

Reply:
This is for the consumer config, it is not the same as create kafka topic and register to kafka broker